marconi-sidechain/src/Marconi/Sidechain/Api/Query/Indexers/MintBurn.hs

```diff
@@ -72,7 +72,6 @@ queryByPolicyAndAssetId env policyId assetId slotNo = do
             (x ^. txMintRowBlockHeaderHash)
             (x ^. txMintRowSlotNo)
             (x ^. txMintRowTxId)
-            Nothing
+            (Just $ C.hashScriptDataBytes $ C.unsafeHashableScriptData $ x ^. txMintRowRedeemerData)
```

Comment:
Maybe not a good idea to use unsafeHashableScriptData.

I think you need to change the MintBurn indexer to work with `HashableScriptData` instead of just `ScriptData`.

Reply:
Could we leave this for a separate ticket? As I'm looking into this then I think this can't be done within cardano-api types as HashableScriptData is made by serialising ScriptData and in the conversion the serialised bytes for the script are already likely not the correct. (This is what I understand the [comment for `unsafeHashableScriptData`](https://github.com/input-output-hk/cardano-api/blob/cb74de6ab4006406003a8f11a24bfb83a32929dc/cardano-api/internal/Cardano/Api/ScriptData.hs#L118-L121) in cardano-api to say)

It seems that the way to get the hash is get it from the ledger transaction within the indexer _before_ it's converted into an cardano-api type [within the getPolicyData](https://github.com/input-output-hk/marconi/blob/7e1b044eab8ba9501a4b3bab9c19695c9926a88c/marconi-chain-index/src/Marconi/ChainIndex/Indexers/MintBurn.hs#L112-L133) (maybe even cardano-ledger-api could help us here, but don't know enough about that).

There are uses of `unsafeHashableScriptData` in the Utxo indexer (and also in the ScriptTx indexer which is where it all probably started. Sorry about that :p) -- when we have the solution we might be able to fix all of them.

Reply:
Right, so the problem with the other `unsafeHashableScriptData` uses is that in the tests all `ScriptData`'s are not parsed from ledger script data but generated directly by the generators and there are no "original" bytes to hash.